src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -295,14 +292,14 @@ export class NotebookService extends Disposable implements INotebookService {
 
 	async getOrCreateNotebookManager(providerId: string, uri: URI): Promise<INotebookManager> {
 		if (!uri) {
-			throw new Error(localize('notebookUriNotDefined', "No URI was passed when creating a notebook manager"));
+			throw new Error(NotebookUriNotDefined);
 		}
 		let uriString = uri.toString();
 		let managers: INotebookManager[] = this._managersMap.get(uriString);
 		// If manager already exists for a given notebook, return it
 		if (managers) {
-			let index = firstIndex(managers, m => m.providerId === providerId);
-			if (index && index >= 0) {
+			let index = managers.findIndex(m => m.providerId === providerId);
+			if (index >= 0) {
```

Comment:
(index && index >= 0) check was a pre-existing bug that I uncovered in testing. With code the way it was even if we had previously created and cached a manager in this._managersMap and managers.findIndex returned 0, we would not use that manager at 0th location but instead create a new one which is not what the code is intending to do.